crates/matrix-sdk-crypto/src/store/mod.rs

```diff
@@ -478,6 +492,55 @@ impl Store {
         self.save_changes(changes).await
     }
 
+    /// Convert the devices and vectors contained in the [`DeviceChanges`] into
+    /// a [`DeviceUpdates`] struct.
+    ///
+    /// The [`DeviceChanges`] will contain vectors of [`ReadOnlyDevice`]s which
+    /// we want to convert to a [`Device`].
+    fn collect_device_updates(
```

Comment:
Why an associated function instead of a free function?

Reply:
Well, it's scoped to the store *module* either way. I have a slight preference towards a free function, but wouldn't mind it merged as-is either.